internal/service/service_test.go

```diff
@@ -0,0 +1,187 @@
+package service_test
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/rs/zerolog"
+	m "github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/andreipimenov/golang-training-2021/internal/mock"
+	"github.com/andreipimenov/golang-training-2021/internal/model"
+	"github.com/andreipimenov/golang-training-2021/internal/service"
+)
+
+const (
+	ticker             = "AAPL"
+	validApiKey        = "123123"
+	invalidTicker      = "DIMMUBORGIR"
+	date               = "2021-07-26"
+	weekendDate        = "2021-07-25"
+	futureDate         = "2099-01-01"
+	dateLayout         = "2006-01-02"
+	invalidApiKey      = ""
+	invalidApiKeyMsg   = "apikey is invalid or missing"
+	invalidTickerMsg   = "failed retrieving data from external API: Invalid API call."
+	tooManyRequestsMsg = "failed retrieving data from external API: Thank you"
+	futureDateMsg      = "failed to find price of"
+	emptyPriceMsg      = "failed to find price of"
+)
+
+// Valid AAPL price for 2021-07-26
+var validPrice = model.Price{
+	Open:  "148.2700",
+	High:  "149.8300",
+	Low:   "147.7000",
+	Close: "148.9900",
+}
+
+type serviceTestSuite struct {
+	suite.Suite
+	s          *service.Service
+	repoMock   *mock.Repository
+	clientMock *mock.HTTPClient
+}
+
+func (suite *serviceTestSuite) SetupTest() {
+	repo := &mock.Repository{}
+	client := &mock.HTTPClient{}
+	s := service.New(&zerolog.Logger{}, repo, validApiKey, client)
+	suite.s = s
+	suite.repoMock = repo
+	suite.clientMock = client
+}
+
+func TestHandler(t *testing.T) {
+	suite.Run(t, new(serviceTestSuite))
+}
+
+func (suite *serviceTestSuite) TestValidPriceFromRepo() {
+	d, err := time.Parse(dateLayout, date)
+	suite.NoError(err)
+
+	suite.repoMock.On("Load", key(ticker, d)).Once().Return(validPrice, true)
+
+	res, err := suite.s.GetPrice(ticker, d)
+
+	suite.NoError(err)
+	suite.Equal(res, &validPrice)
```

Comment:
Equal method has this signature (expected interface{}, actual interface{}, msgAndArgs ...interface{}), that means, you should place expected value on the left, and actual one on the right. It is important for logging errors

Reply:
Fixed